Fixes #30943 - adds additional remote db notes (#1696)
theforeman/theforeman.org: plugins/katello/3.15/user_guide/remote_databases/index.md

    * restore the DBs on remote servers
    * run foreman-installer with right parameters pointing to the databases. It re-configures the databases and start all the services with new DB locations

### Important note:

> Pulpcore installs a separate PostgreSQL database and lacks support for external databases. Databases should be upgraded to avoid conflicts.

Reviewer: ```suggestion
Pulpcore installs a separate PostgreSQL database and lacks support for external databases. This support is introduced in Katello 3.16.

Foreman and Pulpcore require PostgreSQL 10 or newer. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.
```

> Katello 3.15 changes the default database for Pulp from PostgreSQL v9.x to PostgreSQL v12. External databases should be upgraded to avoid conflicts.

Reviewer: :-1: on "changes the default database for Pulp". It gives the impression MongoDB is removed, but it's not. However, it's safe to to say Foreman 2.0 has dropped support for PostgreSQL 9.x and users should upgrade to 12.

I also strongly insist that a separate note is made that Pulp 3 is installed alongside, but that lacks remote database support. Having that would have saved me from answers the same question in support.

    * restore the DBs on remote servers
    * run foreman-installer with right parameters pointing to the databases. It re-configures the databases and start all the services with new DB locations

### Important note:

> * Pulpcore installs a separate PostgreSQL database and lacks support for external databases. This support is introduced in Katello 3.16.

Reviewer: ```suggestion
* Pulpcore installs a separate PostgreSQL database. The installer lacks support for external databases. This support is introduced in Katello 3.16.
```

> * Foreman v2.0 and Pulpcore require PostgreSQL v10+](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures) Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.

Reviewer: ```suggestion
* Foreman ([since 2.0](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures)) and Pulpcore require PostgreSQL 10+. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.
```

> * Foreman ([since 2.0])(https://theforeman.org/manuals/2.0/index.html#Headlinefeatures) and Pulpcore requires PostgresSQL 10+. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.

Reviewer: Not a native speaker, but isn't it plural?
```suggestion
* Foreman ([since 2.0])(https://theforeman.org/manuals/2.0/index.html#Headlinefeatures) and Pulpcore require PostgresSQL 10+. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.
```

Author: You are correct!

    * restore the DBs on remote servers
    * run foreman-installer with right parameters pointing to the databases. It re-configures the databases and start all the services with new DB locations

### Important note:

Pulpcore installs a separate PostgreSQL database and lacks support for external databases. This support is introduced in Katello 3.16.

> Foreman and Pulpcore require PostgreSQL 10 or newer. Databases should be upgraded to avoid conflicts. It is recommended to update to version 12.

Reviewer: In other files it uses bullet points, here it doesn't. Feels a bit inconsistent.

Author: @ekohl After I remove the second line, we no longer have "important notes" but an "important note: I wonder if that formatting should change too.

> Note that [Foreman v2.0 requires PostgreSQL v10+](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures).

Reviewer: You can drop this since it's duplicate from the line above. Perhaps you could add the link there.

    * restore the DBs on remote servers
    * run foreman-installer with right parameters pointing to the databases. It re-configures the databases and start all the services with new DB locations

> ### Important note: Katello 3.15 changes the default database for pulp from postgres v9.x to postgres v12. External databases should be upgraded to avoid conflicts.

Reviewer: The correct way to write it is PostgreSQL. I'd also use Pulp.

```suggestion
### Important note: Katello 3.15 changes the default database for Pulp from PostgreSQL v9.x to PostgreSQL v12. External databases should be upgraded to avoid conflicts.
```

It may also be good to note that Foreman itself recommends upgrading to version 12. https://theforeman.org/manuals/2.0/index.html#Headlinefeatures already mentions it. https://theforeman.org/manuals/2.0/index.html#3.1.1SupportedPlatforms also states you must at least run PostgreSQL 10 for Foreman itself.

This also suggests MongoDB is no longer needed for Pulp, but that's not really true.

> ### Important note: Katello 3.15 changes the default database for Pulp from PostgreSQL v9.x to PostgreSQL v12. External databases should be upgraded to avoid conflicts.

Reviewer: One thing that a lot of 3.15 users ran into was that Pulpcore doesn't support external databases in this version, so you'll always end up with a local database. IMHO it was a mistake to release Pulpcore without external DBs and it would have saved a lot of support questions if we at least documented this.

Author: In that case maybe we should remove "External databases..." and instead point out that 3.15 doesn't support external databases?

Reviewer: You can still use Foreman and Candlepin external databases. The Foreman DB must be updated to at least 10 on Foreman 2.0.
